c_common/models/data_speed_up_packet_gatherer/src/data_speed_up_packet_gatherer.c

```diff
@@ -359,20 +359,42 @@ static inline void set_message_length(const void *end) {
 //! \brief searches through received seq nums and transmits missing ones back
 //! to host for retransmission
 static void process_missing_seq_nums_and_request_retransmission(void) {
+    //! \brief Used to guard access to the received_seq_nums_store from this
+    //!   function; it counts the number of running calls to this function.
+    //!   Access to this variable is only allowed when you have disabled
+    //!   interrupts!
+    static uint access_lock = 0;
+
+    uint sr;
+    sr = spin1_irq_disable();
+    if (++access_lock == 0) {
```

Comment:
Is this correct?  I am looking at this and thinking that if adding 1 to an unsigned integer == 0, then value must have been 0xFFFFFFFF previously...